holodeck/environments.py

```diff
@@ -157,14 +164,19 @@ def step(self, action):
         if not self._initial_reset:
             raise HolodeckException("You must call .reset() before .step()")
 
-        self._agent.act(action)
+        if self._agent is not None:
+            self._agent.act(action)
 
-        self._command_center.handle_buffer()
-
-        self._client.release()
-        self._client.acquire()
+            self._command_center.handle_buffer()
+            self._client.release()
+            self._client.acquire()
+            return self._get_single_state()
 
-        return self._get_single_state()
+        else:
+            self._command_center.handle_buffer()
+            self._client.release()
+            self._client.acquire()
+            return self._get_full_state()
```

Comment:
Should we put the state into a variable so that we can pull the command center code out of the conditional and only have one return statement? This might just be personal preference.